src/SamplingSchemes/exact_enumeration.jl

```diff
@@ -0,0 +1,66 @@
+"""
+    exact_enumeration(lattice::LatticeSystem{G}, cutoff_radii::Tuple{Float64, Float64}, h::LatticeGasHamiltonian) where G
+
+Enumerate all possible configurations of a lattice system and compute the energy of each configuration.
+
+# Arguments
+- `lattice::LatticeSystem{G}`: The (starting) lattice system to enumerate. All possible configurations will be generated from this lattice system.
+- `cutoff_radii::Tuple{Float64, Float64}`: The cutoff radii for the first and second nearest neighbors.
+- `h::LatticeGasHamiltonian`: The lattice gas Hamiltonian.
+
+# Returns
+- `energies::Vector{typeof(0.0u"eV")}`: An array of energies for each configuration.
+- `configurations::Vector{LatticeSystem{G}}`: An array of lattice system configurations for each configuration.
+- `walkers::Vector{LatticeWalker}`: An array of lattice walkers for each configuration.
+"""
+function exact_enumeration(
+    lattice::LatticeSystem{G},
+    cutoff_radii::Tuple{Float64, Float64},
+    h::LatticeGasHamiltonian,
+    ) where G
```

Comment:
Is `G` for `LatticeSystem` similar to `C` for the atomistic systems?

Reply:
Exactly. `G` is a so-called type parameter, with an hint to "geometry".